src/engine/blocks.js

```diff
@@ -75,37 +75,34 @@ Blocks.prototype.getBranch = function (id, branchNum) {
     }
 
     // Empty C-block?
-    if (!(inputName in block.inputs)) return null;
-    return block.inputs[inputName].block;
+    var input = block.inputs[inputName];
+    return (typeof input === 'undefined') ? null : input.block;
 };
 
 /**
  * Get the opcode for a particular block
- * @param {?string} id ID of block to query
+ * @param {?object} block The block to query
  * @return {?string} the opcode corresponding to that block
  */
-Blocks.prototype.getOpcode = function (id) {
-    var block = this._blocks[id];
+Blocks.prototype.getOpcode = function (block) {
```

Comment:
That makes sense. Probably ok to leave for now. Might be a good idea to add a `@todo` note here though.